### ISSUE TITLE
grafana alert's template variables don't work when terms have "."

### ISSUE THREAD

**Issue by houhuijian** [grafana/grafana#48567]
datasource：Elasticsearch for wazuh systems
version：8.3.3
problem：
I will display the value of a label in the annotation, if I set {{ $values.B.Labels }},The output I see is "map[agent.ip=xx.xx.xx.xx]"，but when i set {{ $values.B.Labels.agent.ip }} ,the output is {{ $values.B.Labels.agent.ip }} ,
I think it should be caused by the inclusion of . in terms, so I added \ in front of . and tried using ' ', " ", { }, [], () for terms, but none of them can be displayed normally.
If this is the case, how should I fix it

**Comment by vvotekeb** [comment]
I have the same issue in 8.5.1

**Comment by santihernandezc** [comment]
Hey @houhuijian, this has to do with how the Go templating engine works. In cases like this one, where you have a key that contains a `.` character, you have to use the `index` template function.

Excerpt from the [docs](https://pkg.go.dev/text/template#hdr-Functions):

```
index
	Returns the result of indexing its first argument by the
	following arguments. Thus "index x 1 2 3" is, in Go syntax,
	x[1][2][3]. Each indexed item must be a map, slice, or array.
```

TLDR, simply use `{{ index $values.B.Labels "agent.ip" }}`.

Useful links:
- I found a [Stack Overflow question](https://stackoverflow.com/questions/24267608/how-to-print-the-value-of-a-key-containing-dots) describing a very similar scenario.
- I made an example for you in the Go Playground, you can run it [here](https://go.dev/play/p/rtaaB-F4A3J).

I'm closing this issue, but feel free to reopen it if you have something else to add!

**Comment by houhuijian** [comment]
I have tried it but it didn't solve the problem
